### ISSUE TITLE
Forcing BITCODE_GENERATION_MODE=bitcode breaks some framework configurations?

### ISSUE THREAD

**Issue by eliburke** [Carthage/Carthage#2255]
* `carthage version`: 0.26.2
* `xcodebuild -version`: Xcode 9.1 Build version 9B55
* Are you using `--no-use-binaries`? YES

**Cartfile**
```
github "realm/realm-cocoa" "v3.0.2"
```

**Carthage Command**
```
carthage update --no-use-binaries --platform iOS --configuration Debug realm-cocoa
```

**Carthage Output**
```
"/var/folders/t_/m9rfl27527756l3j41hnjsm40000gn/T/carthage-xcodebuild.bK4qjL.log" 2299L, 1067961C
ld: bitcode bundle could not be generated because 'core/librealm-ios-dbg.a(bptree.o)' was built without full bitcode. All object files and libraries for bitcode must be generated from Xcode Archive or Install build for architecture arm64
```

My goal is to create a debug build of realm-cocoa. My issue started at the confluence of releases of iOS 11, Swift 4, Xcode 9, realm-cocoa 3.0, and carthage .24-.26, so I don't know exactly who to point fingers at. I opened realm/realm-cocoa#5419 to talk with the realm-cocoa devs, and their response was: 
`"We exclude bitcode from the prebuilt core library [...] but Carthage passes BITCODE_GENERATION_MODE=bitcode to xcodebuild, overriding what we have set in the project."`

I know that one of the changes in 0.26 was switching to the archive action. Is that what forces it to enable set BITCODE_GENERATION_MODE=bitcode? Is there a way to revert to the older behavior with a command line switch or project configuration, on a case-by-case basis?

**Comment by jnutting** [comment]
@eliburke, did you ever find a solution to this problem? I'm having the exact same issue.

**Comment by eliburke** [comment]
Unfortunately, no. The Realm folks don't seem to view this as a serious issue to pursue, and I'm not sure if the Carthage folks have noticed this issue. I think (but do not know) that the problem is not with the ENABLE_BITCODE flag,  but rather with BITCODE_GENERATION_MODE. Perhaps you can try your XCODE_CONFIG_FILE technique with that variable?

**Comment by jdhealy** [comment]
> `Carthage passes BITCODE_GENERATION_MODE=bitcode to xcodebuild`

https://github.com/Carthage/Carthage/blob/26e2420d789902088573cf6a74d3441cf76bbc3b/Source/CarthageKit/Xcode.swift#L625-L628

Unless I'm missing something, Carthage passes BITCODE_GENERATION_MODE=bitcode solely in this situation.

> Perhaps you can try your XCODE_CONFIG_FILE technique with [BITCODE_GENERATION_MODE]

Right on, should be one way to avoid the above situation: 

```
tee > no-bitcode.xcconfig <<-'EOF'
ENABLE_BITCODE = NO
BITCODE_GENERATION_MODE = marker
EOF

XCODE_XCONFIG_FILE='no-bitcode.xcconfig' carthage build …
```

Maybe even add `SUPPORTED_PLATFORMS = iphoneos`, if skipping a simulator build is desired.

I overran my per-run CI time limit when actually testing this with Realm, but seemingly should work.

**Comment by ikesyo** [comment]
Should be addressed by #2435.
